autorest/adal/go.mod

```diff
@@ -8,6 +8,9 @@ require (
 	github.com/Azure/go-autorest/autorest/mocks v0.4.1
 	github.com/Azure/go-autorest/logger v0.2.1
 	github.com/Azure/go-autorest/tracing v0.6.0
-	github.com/form3tech-oss/jwt-go v3.2.2+incompatible
+	// NOTE: cannot move to github.com/golang-jwt/jwt/v4 yet, because the main
+	//       github.com/Azure/go-autorest code uses go dep, which is not
+	//       compatible with v2+ go modules that do not use a v<version> directory.
+	github.com/golang-jwt/jwt v3.2.2+incompatible
```

Comment:
I think we can rip the band-aid off now and just delete the gopkg.* files, then update to v4.

Reply:
That'd be great! I wasn't sure if someone / some team internal at Microsoft was depending on this, which is why I tried to keep things as-is.

I can give it a quick try later to create a suitable `go.mod` (and 🤞 hoping `go mod` is cooperating 😂)

Reply:
Oh, hmm... didn't fully realise the "main" module is empty. I wonder if it's better to _add_ a module or to just remove the GoDep files.

Reply:
It should be ok to remove them.  The only reason I left them was to keep back-compat during the transition to modules.  Now that we're several releases in and the world has moved on from `dep` I think it's ok to remove them.

Reply:
Adding a `go.mod` would also mean it has to be renamed to `github.com/Azure/go-autorest/v15`, so perhaps we should just remove GoDep, and consider the repository to be a "collection of modules" without a "main" module 🤔

Reply:
Yes exactly, and is the reason I didn't add one during the initial migration (although, if I had to do it over again, I'd make `/autorest` one module instead of several).